bravado/response.py

```diff
@@ -69,9 +69,10 @@ def result(self, **kwargs):
 
         if self.cancelled():
             raise CancelledError()
-        response = self._request.wait(timeout=timeout)
+        response = self._request.result(timeout=timeout)
         try:
-            response.raise_for_status()
+            if hasattr(response, 'raise_for_status'):
```

Comment:
This should really be part of a specific Response implementation for each HttpClient.

Using the interfaces I describe above it would be easy to do this:

```python

class HttpFuture(object):

    def __init__(self, client, response_handler):
       ...

```

A `ResponseHandler` (currently called `post_receive`) can deal with the differences between each client, and since the client returns the Future, we don't need any branching. It know the correct class to return to handle the response.